lib/ansible/modules/monitoring/zabbix/zabbix_host.py

```diff
@@ -489,7 +489,7 @@ def check_all_properties(self, host_id, host_groups, status, interfaces, templat
 
         # Check whether the visible_name has changed; Zabbix defaults to the technical hostname if not set.
         if visible_name:
-            if host['name'] != visible_name and host['name'] != host_name:
+            if host['name'] != visible_name and host['host'] != visible_name:
```

Comment:
This wouldn't be possible with the patch if lets say _Visible name_ is set to `examplehost1` as it would return OK:
```yaml
      zabbix_host:
        server_url: "{{ zbx_server }}"
        login_user: "{{ zbx_user }}"
        login_password: "{{ zbx_pass }}"
        host_name: "examplehost"
        visible_name: "examplehost"
```

I see no point in checking _Host name_ at all. Zabbix API will feed `host['name']` with actual _Host name_ instead of _Visible name_ if there is none defined so lets try it like this instead (merge line 492 into 491):
```suggestion
            if visible_name and host['name'] != visible_name:
```